src/main/java/software/sigma/internship/dto/FundDto.java

```diff
@@ -1,4 +1,10 @@
 package software.sigma.internship.dto;
 
-public record FundDto(long id, String name, String description, String link) {
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
+public record FundDto(long id,
+                      @NotNull @NotBlank String name,
+                      @NotNull @NotBlank String description,
+                      @NotNull @NotBlank String link) {
```

Comment:
Make a validation for a valid url. I believe you can use URL as field type here, but you have to try and see.